.github/workflows/ci.yaml

```
@@ -70,6 +70,33 @@ jobs:
           cargo package -p hypersync-net-types
           cargo package -p hypersync-client
 
+  # The capnp-generated code is committed rather than built from build.rs, so it
+  # can drift from the schema if someone edits the .capnp file without
+  # regenerating. Regenerate here and fail if the committed code is stale. The
+  # plugin version is pinned exactly: the generated code's bytes depend on the
+  # capnpc-rust version, so a floating "^0.23" would produce spurious diffs.
+  check_capnp_generated:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```

actionlint (1.7.12)

[error] the runner of "actions/checkout@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)